src/libraries/System.Net.Primitives/src/System/Net/IPAddressParser.cs

```diff
@@ -107,6 +111,36 @@ private static unsafe bool TryParseIPv6(ReadOnlySpan<char> ipSpan, Span<ushort>
             return false;
         }
 
+        private static ReadOnlySpan<char> RemoveLeadingZerosFromIPv4(ReadOnlySpan<char> ipSpan)
+        {
+            Span<char> cleanedIp = stackalloc char[ipSpan.Length];
+            int index = 0;
+            int segmentStart = 0;
```

Comment:
This method is incorrect, and will never work in this form. It returns stack allocated memory out of a method. This is a common mistake in C, and the span lifetime rule of C# blocks this.

Instead, the `TryParseIpv4` method should be updated to accept leading zeros.
Note that the method is currently not optimial.